### PR TITLE
Add proper support for list mutation

### LINKED ISSUE #95: Add proper support for list mutation
I think we already have frontend support for this. May need to properly validate in semantic analysis if we properly catch mutations to readonly lists at compile time. In runtime we should be able to use the same logic as the compile time to validate the element type (this may be less efficient but should be correct; I think it is worth it to revisit this when this becomes a perf issue)

### PR DESCRIPTION
**Pull request by heshanpadmasiri**
## Purpose
$subject
Resolves #95

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

 

## Summary by CodeRabbit

* **Documentation**
  * Added production/debug build instructions and detailed profiling guide with examples.

* **New Features**
  * Debug-mode profiling CLI flags and start/stop hooks (server address configurable).
  * Improved list construction and runtime list behavior visible to programs.
  * Cycle-safe, improved printing/formatting for complex or nested values.

* **Chores**
  * Expanded example/test corpus demonstrating list behaviors and edge cases.

### CODE REVIEW

File: bir/bir_gen.go [hunk 1]
```diff
@@ -541,18 +542,43 @@ func typeConversionExpression(ctx *stmtContext, curBB *BIRBasicBlock, expr *ast.
...
-	constantLoad.Value = int64(-1)
+	constantLoad.Value = int64(len(expr.Exprs))
 	constantLoad.LhsOp = sizeOperand
 	bb.Instructions = append(bb.Instructions, constantLoad)
 
+	lat := expr.AtomicType
+	for i := len(expr.Exprs); i < lat.Members.FixedLength; i++ {
+		ty := lat.MemberAt(i)
+		fillerVal := values.DefaultValueForType(ty)
+		if fillerVal == values.NeverValue {
```
Comment (warunalakshitha) [inline bir/bir_gen.go]: IMO we do not need to check this. It will break at runtime if it happens (should never)

File: cli/cmd/prof_debug.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
+//go:build debug
```
Comment (warunalakshitha) [inline cli/cmd/prof_debug.go]: Should we add this before header or after header?

File: runtime/internal/exec/non_terminators.go [hunk 3]
```diff
@@ -101,15 +100,18 @@ func castValue(value any, targetType semtypes.SemType) any {
...
-func resizeArrayIfNeeded(arrPtr *[]any, arr []any, idx int) []any {
-	if idx >= len(arr) {
-		newArr := make([]any, idx+1)
-		copy(newArr, arr)
-		*arrPtr = newArr
-		return newArr
-	}
-	return arr
-}
+// func valueType(value any) semtypes.SemType {
```
Comment (warunalakshitha) [inline runtime/internal/exec/non_terminators.go]: We can remove this?

File: values/list.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+// under the License.
+
+package values
+
+import (
+	"ballerina-lang-go/semtypes"
+	"math"
+)
+
+type List struct {
```
Comment (warunalakshitha) [inline values/list.go:24]: Should we use Array or List as struct name?

File: bir/bir_gen.go (hunk 1 above)
Comment (heshanpadmasiri) [reply bir/bir_gen.go]: Check removed

File: cli/cmd/prof_debug.go (hunk 2 above)
Comment (heshanpadmasiri) [reply cli/cmd/prof_debug.go]: I don't have a strong opinion about either approach I think the restriction is there can't be code before it. Moved to after license in both files

File: values/list.go (hunk 4 above)
Comment (heshanpadmasiri) [reply values/list.go:24]: I prefer List because that matches the terminology of the spec and internally we use terms Array and Tuple in some places for subsets of List

File: corpus/integration_test.go [hunk 5]
```diff
@@ -56,7 +58,9 @@ var (
 	errorRegex  = regexp.MustCompile(`//\s*@error`)
 
 	// Skip tests that cause unrecoverable Go runtime errors
-	skipTestsMap = makeSkipTestsMap([]string{})
+	skipTestsMap = makeSkipTestsMap([]string{
+		"subset3/03-list/24-v.bal",
```
Comment (heshanpadmasiri) [inline corpus/integration_test.go:62]: @warunalakshitha this is the test case where I mentioned we have an issue with how we resize the underlying array (copying instead of appending)

Comment (warunalakshitha) [reply corpus/integration_test.go:62]: Will check on this.

**Review by warunalakshitha — APPROVED** [review]
LGTM
